frontend/src/api/prometheus/kservePerformanceMetrics.ts

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { KserveMetricGraphDefinition } from '~/concepts/metrics/kserve/types';
+import { defaultResponsePredicate } from '~/api/prometheus/usePrometheusQueryRange';
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import { TimeframeTitle } from '~/concepts/metrics/types';
+import useQueryRangeResourceData from '~/api/prometheus/useQueryRangeResourceData';
+import { PendingContextResourceData, PrometheusQueryRangeResultValue } from '~/types';
+
+type RequestCountData = {
+  data: {
+    successCount: PendingContextResourceData<PrometheusQueryRangeResultValue>;
+    failedCount: PendingContextResourceData<PrometheusQueryRangeResultValue>;
+  };
+  refreshAll: () => void;
+};
+
+export const useFetchKserveRequestCountData = (
+  metricsDef: KserveMetricGraphDefinition,
+  timeframe: TimeframeTitle,
+  endInMs: number,
+  namespace: string,
+): RequestCountData => {
+  const active = useIsAreaAvailable(SupportedArea.K_SERVE_METRICS).status;
+
+  //TODO: Necessary due to bug on backend - must be removed before release.
+  const successQuery = metricsDef.queries[0].query; //.replace('[${rate_interval}]', '[5m]');
```

Comment:
Is the commented-out code part of the TODO?

Reply:
The commented out code was effectively me fulfilling the TODO... just after testing it, rather than going back and removing the commented code and the todo I moved onto the next thing d'oh!